File: app/src/main/java/me/imzack/app/end/injector/module/TypeCreationPresenterModule.java
package me.imzack.app.end.injector.module;

import me.imzack.app.end.view.contract.TypeCreationViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class TypeCreationPresenterModule {

    private final TypeCreationViewContract mTypeCreationViewContract;

    public TypeCreationPresenterModule(TypeCreationViewContract typeCreationViewContract) {
        mTypeCreationViewContract = typeCreationViewContract;
    }

    @Provides
    TypeCreationViewContract provideCreateTypeViewContract() {
        return mTypeCreationViewContract;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/TypeCreationPresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.TypeCreationViewContract

@Module
class TypeCreationPresenterModule(private val mTypeCreationViewContract: TypeCreationViewContract) {

    @Provides
    fun provideCreateTypeViewContract() = mTypeCreationViewContract
}
